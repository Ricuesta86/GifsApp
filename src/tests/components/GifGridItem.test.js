import React from 'react';
import { shallow } from 'enzyme';    
import GifGridItem from '../../components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
    
    const title = 'Hola mundo';
    const url = 'https://localhost/algo.jpg';

    const wrapper = shallow( <GifGridItem 
            title={ title }
            url = { url }
            /> );

    test('debe mostrar el componente correctamente ', () => {



        expect( wrapper ).toMatchSnapshot();
    })
    
    test('debe tener un pÃ¡rrafo con el title', () => {
        
        const p = wrapper.find( 'p' );
        expect( p.text().trim() ).toBe( title );

    })
    
    test('debe tener la imagen igual a la url y el alt al title', () => {
        
        const img = wrapper.find('img');
        // console.log(img.props());

        expect( img.props().src.trim() ).toBe(url);
        expect( img.props().alt.trim() ).toBe(title);

    })

    test('debe tener la clase animate__fadeInUp', () => {
        
        const div = wrapper.find('div');
        // console.log( div.prop('className') );
        const className = div.prop('className');

        expect(className.includes('animate__fadeInUp')).toBe(true);
    })
    
    
    
    
})
