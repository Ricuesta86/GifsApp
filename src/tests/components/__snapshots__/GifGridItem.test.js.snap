// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> debe mostrar el componente correctamente  1`] = `
<div
  className="card animate__animated animate__fadeInUp"
>
  <img
    alt="Hola mundo"
    src="https://localhost/algo.jpg"
  />
  <p>
    Hola mundo
  </p>
</div>
`;
